test/smoke_test.js

```diff
@@ -129,10 +142,75 @@ const path = require("path");
     return page.waitForSelector("header .login-button", { visible: true });
   });
 
+  // Prep for assets check
+  const anyStylesheetEl = await page.$('link[href][rel="stylesheet"]');
+  const anyAssetPath = await page.evaluate(
+    el => el.getAttribute("href"),
+    anyStylesheetEl
+  );
+  if (!anyAssetPath) {
+    return console.log("FAILED - could not retrieve an asset path");
+  }
+
   if (process.env.LOGIN_AT_BEGINNING) {
     await login();
   }
 
+  function checkNoCookies(testPath) {
+    return async function() {
+      const priorCookie = await getCookie("_t", page);
+      const testURL = new URL(testPath, url);
+
+      await page.setCookie({
+        name: "_t",
+        value: "invalid_auth_token",
+        url: url,
+        domain: priorCookie.domain,
+        path: priorCookie.path,
+        expires: priorCookie.expires,
+        httpOnly: priorCookie.httpOnly,
+        secure: priorCookie.secure,
+        session: priorCookie.session,
+        sameSite: priorCookie.sameSite
+      });
+      const badCookie = await getCookie("_t", page);
+      if (badCookie.value !== "invalid_auth_token") {
+        throw "FAILED - could not set cookie";
+      }
+
+      await page.goto(testURL);
+
+      const newCookie = await getCookie("_t", page);
+      if (newCookie === null || newCookie.value !== "invalid_auth_token") {
+        throw "FAILED - Cookie was modified while fetching " + testPath;
+      }
+
+      await page.setCookie(priorCookie);
+    };
+  }
+
+  await exec(
+    `assets do not set cookies (${anyAssetPath})`,
+    checkNoCookies(anyAssetPath || "/assets/stylesheets/bogus.css")
+  );
+  await exec(
+    "service-worker.js does not set cookies",
+    checkNoCookies("/service-worker.js")
+  );
+  await exec("cookie check works", async () => {
+    const fn = checkNoCookies("/about");
+    let failure = false;
+    try {
+      await fn();
+      failure = true;
+    } catch (e) {
+      // pass
```

Comment:
Might be good to update the comment here a little more explicitly say that it passes because we expect checkNoCookies to fail+throw; this may be difficult to reason about/read through later down the road otherwise.